main.tf

```diff
@@ -29,10 +10,14 @@ resource "google_project" "project" {
   labels = var.deployment_project_labels
 }
 
+data "google_project" "project" {
+  project_id = var.deployment_project_id
+}
```

Comment:
Should we add a depends_on here for google_project.project?  It looks like all the other resources are expecting this data object to return the project now. In the case where it's net new and you wanted this module to create the project. I'm not sure if the data block will try to evaluate before the project is created.